packages/js/product-editor/src/components/editor/editor.tsx

```diff
@@ -49,10 +50,15 @@ type EditorProps = {
 	settings: ProductEditorSettings | undefined;
 };
 
+const EDITOR_ID = 'product-block-editor';
 export function Editor( { product, settings }: EditorProps ) {
 	const [ selectedTab, setSelectedTab ] = useState< string | null >( null );
 
-	const updatedLayoutContext = useExtendLayout( 'product-block-editor' );
+	const updatedLayoutContext = useExtendLayout( EDITOR_ID );
+
+	useEffect( () => {
+		document.body.classList.add( EDITOR_ID + '-page' );
```

Comment:
I resorted to this to be able to target the `woocommerce-layout__header` correctly, given WordPress does something similar in the post/page editor pages.

The alternative would be to actually not render anything in the Header when on a product page. But this required specific logic within the Header component around the product pages, which seemed not great.